### PR TITLE
Refine MultiFormatReader results filtering and apply C++20 `erase_if`

### PR DESCRIPTION
**Pull request by Doekin**

### CODE REVIEW

**Comment by axxel** [comment]
May I ask what your intention with this PR is? Just cosmetic or did you spot the 2 year old bug that you happen to fix with this? :)

**Comment by Doekin** [comment]
I am currently learning how to read barcodes fast and accurately, and happened to stumble upon this.

**Comment by axxel** [comment]
Would you do me the favor of splitting this into a first commit that just fixes the bug and then if you feel inclined, a second that introduces the `erase_if()` but not only here but for all 7 occurances of `std::remove()` throughout the code base?

**Comment by Doekin** [comment]
Certainly, I'll take care of that.

File: core/src/BarcodeFormat.cpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
-* Copyright 2016 Nu-book Inc.
-* Copyright 2016 ZXing authors
-*/
+ * Copyright 2016 Nu-book Inc.
```
Comment (axxel) [inline core/src/BarcodeFormat.cpp:2]: Looks awesome, but can you please revert those unrelated "* ..." -> " *..." whitespace changes?

Comment (Doekin) [reply core/src/BarcodeFormat.cpp]: OK

**Comment by axxel** [comment]
Thanks for the effort and your cooperation.

**Pull request merged**